packages/story-editor/src/app/media/utils/useGetResourceFromUrl.js

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import {
+  createResource,
+  getFileExtFromUrl,
+  getFileNameFromUrl,
+  getFirstFrameOfVideo,
+  getImageDimensions,
+  getVideoDimensions,
+  getVideoLength,
+  hasVideoGotAudio,
+} from '@web-stories-wp/media';
+
+/**
+ * Internal dependencies
+ */
+import { useConfig } from '../../config';
+import { getPosterName, useUploadVideoFrame } from '.';
+
+const EXT_MIME_TYPES = {
+  jpg: 'image/jpeg',
+  gif: 'image/gif',
+  jpe: 'image/jpe',
+  jpeg: 'image/jpg',
+  m4v: 'video/mp4',
+  mp4: 'video/mp4',
+  png: 'image/png',
+  webm: 'video/webm',
+  webp: 'image/webp',
+};
+
+function useGetResourceFromUrl() {
```

Comment:
All other `getResourceFrom` utils are simple functions, but this is a hook.

Could you split this up into a hook and a separate `getResourceFromUrl` function?

Reply:
Hm, I'm not sure what would be the best way to split it up like this without the server side validation for the type. If we pass just the url to the `getResourceFromUrl` then the first thing we need to do is check if the extension is valid and throw an error if not. We can't check that without `allowedMimeTypes` though (which uses `useConfig`). 

I suppose we could use the hardcoded `EXT_MIME_TYPES` for now? Thoughts?

Reply:
The `uploadVideoPoster()` part and the mime type check could be done in `useInsert`/`onInsert`.

The mime type check can be done server-side only later on anyway.

Then `getResourceFromUrl` can become a simple function, perhaps taking a signature like `getResourceFromUrl(url, type)`

Reply:
Then it won't fully be getting the resource from the `url` (as in case of the other utils) -- but if that's not important then yes -- we can create a simple function that creates the resource from the url + skips the poster 👍